### ISSUE TITLE
Unresolved aliases

### ISSUE THREAD

**Issue by femtomc** [mkdocstrings/python#59]
**Describe the bug**
I run `mkdocs` with `mkdocstrings` and `griffe` attempts to resolve aliases, but fails to do so.

**Expected behavior**
I expect `griff` and `mkdocstrings_handlers` to resolve aliases.

**Screenshots**

<img width="710" alt="image" src="https://user-images.githubusercontent.com/34410036/216512158-902560eb-be51-4a49-a0d1-cf370b560b40.png">

Eventually, `mkdocs` throws:

```
  File "/Users/mccoybecker/miniconda3/envs/py311/lib/python3.11/site-packages/jinja2/loaders.py", line 126, in load
    source, filename, uptodate = self.get_source(environment, name)
                                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/mccoybecker/miniconda3/envs/py311/lib/python3.11/site-packages/jinja2/loaders.py", line 218, in get_source
    raise TemplateNotFound(template)
jinja2.exceptions.TemplateNotFound: alias.html
(py311)
```

Does anyone know the cause of inability to resolve aliases?

**Comment by pawamoy** [comment]
Hello, thanks for the report!

Please provide a public repository that allows to replicate the issue.
This `TemplateNotFound: alias.html` error is usually caused by compiled modules that don't set the right `__module__` attribute on their members.

**Comment by femtomc** [comment]
Ah interesting. The library I’m developing is currently closed source — but I’ve organized the source code with aliases and an internal structure where I have a _src dir for internals, and then a bunch of public API modules.

@pawamoy is there another issue where you discovered the above comment? (E.g. module sunder setting incorrectly?)

**Comment by pawamoy** [comment]
Yes, sorry! Here are the links :slightly_smiling_face: 

- https://github.com/mkdocstrings/mkdocstrings/issues/514
- https://github.com/mkdocstrings/mkdocstrings/issues/451
- - https://github.com/mkdocstrings/mkdocstrings/issues/404

**Comment by femtomc** [comment]
@pawamoy By the way, here's an example of the pattern of exports that I'm using:

In: `core.py` (public API)

```python
from my_lib._src.core import mask
from my_lib._src.core import none_select
from my_lib._src.core import val_chm
from my_lib._src.core import value_choice_map

__all__ = ["mask", "none_select", "val_chm", "value_choice_map"]
```

So e.g. this toplevel module `core` imports from the private `_src.core` -- and then, inside my documentation file `core.md`:

```
# Core

## Core datatypes

::: my_lib.core.val_chm
```

Is there anything about this pattern that seems like it would imply the problem you've commented on above?

**Comment by pawamoy** [comment]
> Is there anything about this pattern that seems like it would imply the problem you've commented on above?

Nope, this is 100% correct and supported. Do you happen to have compiled modules in your project?

**Comment by femtomc** [comment]
I do not actually know how to check if I have compiled modules in my project :) Haha. But if you point me to some docs I can easily check and return back.

**Comment by pawamoy** [comment]
Oh, what I mean is simply: does your project uses C extension, Cython, Maturin, Rust, etc., that is then compiled to binary Python modules (.so files)?

**Comment by femtomc** [comment]
No, it doesn't.

Okay, I'll make an MRE! Thank you.

**Comment by femtomc** [comment]
@pawamoy Would Python versioning make a difference here? I'm using `3.11` currently.

**Comment by femtomc** [comment]
Re -- I attempted to make an MWE with the same directory pattern as my repo (no luck, it works!) -- I'll try a few more things today to try and replicate the error.

If all else fails, I can invite you to collaborate on the repo and that can give you a reproducer.

**Comment by pawamoy** [comment]
Yes, accessing the repo temporarily would be great. Just make sure your company/employer is OK with that :)

**Comment by femtomc** [comment]
@pawamoy Done :) Please don't share for now (we're open sourcing in the future).

Let me know if you need anything or want me to do a Zoom debug session / etc. Thank you so much for your help with this!

**Comment by pawamoy** [comment]
OK I was able to test things in your repo, and identified the cause of the issue.

- I see that you're using wildcard imports
- Wildcard imports are not perfectly handled by Griffe (they're hard to get right)
- Griffe suffers from a design flaw where a module can't have both a member and a submodule of the same name. On the todo list, but didn't get time yet.
- Griffe wrongly things that `your_lib._src.core` is `jax.core` because of wildcard imports, and since `jax` is not loaded, resolving `your_lib.core.val_chm` fails.

Possible workarounds:

- stop using wildcard imports (they are evil :grinning:)
- or at least define `__all__` variables when you wildcard import things, this will help Griffe

**Comment by femtomc** [comment]
Thank you so much! I’ll make these changes.
